L1Trigger/L1THGCal/src/be_algorithms/HGCalShowerShape.cc

```diff
@@ -294,18 +300,27 @@ float HGCalShowerShape::sigmaPhiPhiMax(const l1t::HGCalMulticluster& c3d) const
 
 float HGCalShowerShape::eMax(const l1t::HGCalMulticluster& c3d) const {
   
-   const edm::PtrVector<l1t::HGCalCluster>& clustersPtrs = c3d.constituents();
+  std::unordered_map<int, float> layer_energy;
 
-   float EMax=0;   
+  const edm::PtrVector<l1t::HGCalCluster>& clustersPtrs = c3d.constituents();
+
+  for(const auto& clu : clustersPtrs){
+    
+    int layer = HGC_layer(clu->subdetId(),clu->layer());        
+    layer_energy[layer] += clu->energy();
```

Comment:
The layer energy should first be initialized to 0

Reply:
Should it?
For a map element, if there is no key, it is initialized in principle with the default constructor of the class (in that case float()=0).

Reply:
Ah, right, that's correct. So it is ok as it is.